### ISSUE TITLE
Undefined index: query (class-amp-youtube-embed.php in AMP_YouTube_Embed_Handler::get_video_id_from_url)

### ISSUE THREAD

**Issue by archon810** [ampproject/amp-wp#3348]
If YouTube is embedded in this format: https://www.youtube.com/embed/xS3yL5kR02 and not the usual ?v= method, the following notice will be triggered:

```
ErrorException: Notice: Undefined index: query
#12 /wp-content/plugins/amp/includes/embeds/class-amp-youtube-embed.php(166): AMP_YouTube_Embed_Handler::get_video_id_from_url
#11 /wp-content/plugins/amp/includes/embeds/class-amp-youtube-embed.php(84): AMP_YouTube_Embed_Handler::shortcode
#10 /wp-content/plugins/amp/includes/embeds/class-amp-youtube-embed.php(106): AMP_YouTube_Embed_Handler::oembed
#9 /wordpress/wp-includes/class-wp-embed.php(173): WP_Embed::shortcode
#8 /wordpress/wp-includes/class-wp-embed.php(427): WP_Embed::autoembed_callback
#7 [internal](0): preg_replace_callback
#6 /wordpress/wp-includes/class-wp-embed.php(409): WP_Embed::autoembed
#5 /wordpress/wp-includes/class-wp-hook.php(286): WP_Hook::apply_filters
#4 /wordpress/wp-includes/plugin.php(208): apply_filters
#3 /wp-content/themes/ap2/amp/single.php(12): include
#2 /wordpress/wp-includes/template-loader.php(78): require_once
#1 /wordpress/wp-blog-header.php(19): require
#0 /index.php(23): null
```

I'm using v1.2.2 of the AMP WP plugin.

**Comment by archon810** [comment]
We have 39 such pages currently: https://www.androidpolice.com/?s=https%3A%2F%2Fwww.youtube.com%2Fembed%2F.

**Comment by westonruter** [comment]
I am seeing that notice as well. However, I am not seeing such URLs resulting in video players in the non-AMP version. How are you embedding such a URL into a post and are you seeing the video render properly on the non-AMP version?

In short, such URLs are technically not correct. The URL is intended to be used as the `src` for an `iframe` rather than the oEmbed URL.

In any case, the notice needs to be prevented. But I'm not sure if that means it should be resulting in an actual YouTube video being rendered, or if it is a failure of user input.

**Comment by archon810** [comment]
I looked into it further, and here's what happens.

- If you paste the url directly into tinymce, WP will not convert it on the spot
- However, if you do this: `[embed]https://www.youtube.com/embed/tZBk_Hb-X0w[/embed]`, it will embed correctly

**Comment by archon810** [comment]
Actually, try https://www.youtube.com/embed/5O-hLjRiBoY. This switches to the proper oembed for me:

![gif](https://user-images.githubusercontent.com/469543/65645790-c2ad8a00-dfad-11e9-9d19-99a198c9979e.gif)

**Comment by pierlon** [comment]
I can no longer replicate this issue as of v1.4.1-alpha (841273327f8849f85df04f144d2810c9ece964b2). [The handler for Youtube embeds](https://github.com/WordPress/wordpress-develop/blob/db4f746b4aa13ce87b065038c8dfea6f7dc9253b/src/wp-includes/embed.php#L201) in WP Core correctly translates it into an embeddable URL and thus the query is correctly parsed.

**Comment by westonruter** [comment]
> I can reproduce this issue now.

I moved this back to _In Progress_. Is that going to be part of #3358 or would it be in a separate PR?

**Comment by pierlon** [comment]
It will be fixed in #3558.

**Comment by csossi** [comment]
Verified in QA
